edx_toggles/tests/test_waffle.py

```diff
@@ -0,0 +1,37 @@
+"""
+Unit tests for waffle classes.
+"""
+
+from django.test import TestCase
+
+# TODO import from edx_toggles.toggles once we remove the legacy classes from the exposed API
+from edx_toggles.toggles.internal.waffle.flag import WaffleFlag
+from edx_toggles.toggles.internal.waffle.switch import WaffleSwitch
+
+
+class WaffleFlagTests(TestCase):
+    """
+    WaffleFlag tests.
+    """
+
+    def test_name_validation(self):
+        WaffleFlag("namespaced.name", module_name="module1")
+        self.assertRaises(
+            ValueError, WaffleFlag, "non_namespaced", module_name="module1"
+        )
+
+
+class WaffleSwitchTest(TestCase):
+    """
+    WaffleSwitch tests.
+    """
+
+    def test_name_validation(self):
+        WaffleSwitch("namespaced.name", module_name="module1")
+        self.assertRaises(
+            ValueError, WaffleSwitch, "non_namespaced", module_name="module1"
+        )
+
+    def test_constructor(self):
```

Comment:
Should this be on WaffleFlag as well?

Reply:
No, because they use the same constructor and we want to avoid too many tests crashing at the same time for the same reason.

Reply:
Maybe a name like `test_base_constructor` or a comment would make that more clear.

Reply:
I'll move this to test the base waffle class.